nixos/modules/system/boot/clevis.nix

```diff
@@ -0,0 +1,97 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.boot.initrd.clevis;
+  supportedFs = [ "zfs" "bcachefs" ];
+in
+{
+
+  meta.maintainers = with maintainers; [ julienmalka camillemndn ];
+
+  options = {
+    boot.initrd.clevis.enable = mkEnableOption (lib.mdDoc "Clevis in initrd");
+
+
+    boot.initrd.clevis.package = mkOption {
+      type = types.package;
+      default = pkgs.clevis;
+      defaultText = "pkgs.clevis";
+      description = lib.mdDoc "Clevis package";
+    };
+
+    boot.initrd.clevis.devices = mkOption {
+      description = "Encrypted devices that need to be unlocked at boot using Clevis";
+      default = { };
+      type = types.attrsOf (types.submodule ({
+        options.secretFile = mkOption {
+          description = lib.mdDoc "Clevis JWE secret file used to decrypt the device at boot";
+          type = types.path;
+        };
+      }));
+    };
+
+    boot.initrd.clevis.useTang = mkOption {
+      description = "Wether the clevis secret used to decrypt the devices are binded to a Tang server";
+      default = false;
+      type = types.bool;
+    };
```

Comment:
Does a tang server have to be running in initrd for this? You haven't added tang to initrd.

Reply:
So this option was just to trigger a warning to remind people that if they use a tang server they have to setup networking in initrd. The tang server is hosted on another machine. After consideration, I think I'll just remove it.

Reply:
Ah sorry no it's also useful to make some systemd units wait for networking if we use a remote tang server so probably wise to keep it.